Restore the caravan on AppVeyor 🐫

dra27 opened the pull request:
Restores the smoke-testing emoji temporarily reverted in #13069.

- [x] Upstream Cygwin release fixing [reported 3.5.0 regression](https://cygwin.com/pipermail/cygwin/2024-April/255807.html); testable by checking the [Cygwin main page](https://cygwin.com) and seeing if "The most recent version of the Cygwin DLL is x.y.z" shows x.y.z >= 3.6.0. _3.5.4 released 25-Aug-2024 with this fix_
- [x] AppVeyor images updated to be using that release; testable by checking [the deployment history](https://www.appveyor.com/updates/) for an update made after the Cygwin release. _See [4 Sep 2024](https://www.appveyor.com/updates/2024/09/04/)_

dra27 committed:
Allow manually forcing AppVeyor Cygwin upgrade

dra27 committed:
Revert "Work-around Cygwin 3.5.x regression"

This reverts commit 6b764e6fcd19856ec2bd9fdb41ce0edbac053bf2.

dra27 commented:
The issue has been fixed upstream (see https://github.com/cygwin/cygwin/commit/579064bf4d408e99ed7556f36a3050c7ee99dee6 🙂)

gasche commented:
The AppVeyor CI passes now. Should we merge this?

dra27 commented:
Not yet - it’s manually installing a pre-release Cygwin DLL… once it’s been released and AppVeyor’s updated, I’ll blow away the head commit and we can merge

shindere commented:
Happy to approve / merge as soon as this is ready.

nojb commented:
> Not yet - it’s manually installing a pre-release Cygwin DLL… once it’s been released and AppVeyor’s updated, I’ll blow away the head commit and we can merge

Two months later, what's the situation?

dra27 commented:
There is no change, as there hasn't been a Cygwin release - but I updated the two tasks with details on how to verify that (and rebased to remove the conflict, although CI will clearly fail)

dra27 commented:
Assuming AppVeyor passes, this should (finally) be good to go!

nojb commented:
CI is green. Merging!
